sockeye/inference.py

```diff
@@ -759,16 +791,19 @@ def translate(self, trans_inputs: List[TranslatorInput], fill_up_batches: bool =
                 translated_chunks.append(IndexedTranslation(input_idx=trans_input_idx, chunk_idx=0,
                                                             translation=empty_translation(add_nbest=(self.nbest_size > 1))))
             else:
-                if len(trans_input.tokens) > self.max_input_length:
+                max_input_length = self.max_input_length
+                #take length of source prefix into account while chunking
+                max_input_length = max_input_length if not trans_input.source_prefix else max_input_length - len(trans_input.source_prefix.split())
```

Comment:
like said above, I think we should make `source_prefix` actually `List[Tokens]` so you don't have to split here and in other places.

Reply:
Now I use `Tokens`, which is a `List[Str]`. Yeah no need to do `split`() here and in other places.